### PR TITLE
Feature/#145751723/restful api endpoints

### PR DESCRIPTION
**Pull request by natalie-elizabeth**
[#](url) 145751723 Make the API endpoints RESTful

**What does this PR do?**
Makes the API endpoints of reaction restful using nimble:restivus.

**Description of Task to be completed?**
Extend Meteor DDP methods and provide restful API.

**What are the relevant pivotal tracker stories?**
#145751723

**screenshots**
This gets all items contained products
image

For a specific item id, you include the id after products in the url : - localhost:3000/api/products/${id}
image

### CODE REVIEW

File: server/api/endpoints/accounts.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+          data: {
+            message: "Account creation unsuccessful"
+          }
+
+
+        };
+      }
+    }
+  });
+  Api.addRoute("accounts/id", { authRequired: false }, {
```
Comment (gngchrs) [inline server/api/endpoints/accounts.js]: `accounts/:id` 🤷‍♂️

File: server/api/endpoints/inventory.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+          status: "fail",
+          message: "Update failed"
+        };
+      }
+    }
+  });
+
+  Api.addRoute("inventory/:id", { authRequired: false }, {
+    get: () => {
+      return inventory.findOne().fetch();
```
Comment (gngchrs) [inline server/api/endpoints/inventory.js]: `.findOne(this.urlParams.id)`

File: server/api/endpoints/orders.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        return {
+          status: "fail",
+          message: "Order unsuccessful"
+        };
+      }
+    }
+  });
+  Api.addRoute("orders/:id", { authRequired: false }, {
+    get: () => {
+      return Orders.findOne().fetch();
```
Comment (gngchrs) [inline server/api/endpoints/orders.js]: ditto: see above comment

File: server/api/endpoints/products.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        return {
+          status: "fail",
+          message: "Product failed to add"
+        };
+      }
+    }
+  });
+  Api.addRoute("products/:id", { authRequired: false }, {
+    get: () => {
+      return Products.findOne().fetch();
```
Comment (gngchrs) [inline server/api/endpoints/products.js]: ditto: see above comment

File: server/api/endpoints/inventory.js (hunk 2 above)
Comment (gngchrs) [reply server/api/endpoints/inventory.js]: .find({_id: this.urlParams.id})
.find({_id: ObjectId(this.urlParams.id)})

or check if `this.urlParams.id` has a value in the first place.

**Review by natalie-elizabeth — COMMENTED** [review]
I have implemented the changes that were required. This final commit contains all the changes

File: server/api/endpoints/cart.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  Api.addRoute("cart", { authRequired: false }, {
+    get: () => {
+      return Cart.find().fetch();
+    },
+    post: {
+      action: () => {
+        if (Cart.insert(this.request.body)) {
+          return {
+            status: "success",
+            statusCode: 200,
```
Comment (johnkariuki) [inline server/api/endpoints/cart.js:15]: Replace the hardcoded http-status codes with this https://www.npmjs.com/package/http-status

**Review by johnkariuki — CHANGES_REQUESTED** [review]

**Pull request closed without merging**